src/chatd.cpp

```diff
@@ -5196,12 +5213,18 @@ void Chat::onUserLeave(Id userid)
         // JOIN -1 for own user, but API response might have cleared the peer list already. In that case,
         // chatd's removal needs to clear the peer list again, since remaining peers would have been
         // added as moderators
+        bool commitEach = mChatdClient.mKarereClient->commitEach();
+        mChatdClient.mKarereClient->setCommitMode(false);
         for (auto &it: mUsers)
         {
             CALL_CRYPTO(onUserLeave, it);
             CALL_LISTENER(onUserLeave, it);
         }
         mUsers.clear();
+        if (commitEach)
+        {
+            mChatdClient.mKarereClient->setCommitMode(true);
+        }
```

Comment:
No need to check previous state. If it has not changed, it does nothing.
```suggestion cpp
            mChatdClient.mKarereClient->setCommitMode(commitEach);
```